src/App.tsx

```diff
@@ -1,61 +1,116 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId: number): User | null => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+export const todosList: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App : React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>(todosList);
+  const [titleName, setTitleName] = useState('');
+  const [selectedUserId, SetSelectedUserID] = useState(0);
+  const [hasTitleError, SetTitleError] = useState(false);
+  const [hasUserError, SetUserError] = useState(false);
+
+  const idForNewTodo = Math.max(...todos.map(todo => todo.id));
+
+  const handleTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitleName(event.target.value);
+    SetTitleError(false);
+  };
+
+  const handleUser = (event: ChangeEvent<HTMLSelectElement>) => {
+    SetSelectedUserID(+event.target.value);
+    SetUserError(false);
+  };
+
+  const resetForm = () => {
+    setTitleName('');
+    SetSelectedUserID(0);
+  };
+
+  const handleFormSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    SetTitleError(!titleName);
+    SetUserError(!selectedUserId);
+
+    const newTodo: Todo = {
+      id: idForNewTodo + 1,
```

Comment:
Ok, I see that you add 1, but this name confused me. Add 1 in the assignment or change the name